experimental/packages/opentelemetry-sdk-metrics-base/src/export/ConsoleMetricExporter.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ExportResult, ExportResultCode } from '@opentelemetry/core';
+import { InstrumentType } from '../InstrumentDescriptor';
+import { AggregationTemporality } from './AggregationTemporality';
+import { ResourceMetrics, DataPointType } from './MetricData';
+import { PushMetricExporter } from './MetricExporter';
+
+/* eslint-disable no-console */
+export class ConsoleMetricExporter implements PushMetricExporter {
+  protected _shutdown = false;
+
+  export(metrics: ResourceMetrics, resultCallback: (result: ExportResult) => void): void {
+    return ConsoleMetricExporter._sendMetrics(metrics, resultCallback);
+  }
+
+  async forceFlush() {}
+
+  selectAggregationTemporality(_instrumentType: InstrumentType): AggregationTemporality {
+    return AggregationTemporality.CUMULATIVE;
+  }
+
+  shutdown(): Promise<void> {
+    this._shutdown = true;
+    return Promise.resolve();
+  }
+
+  private static _sendMetrics(metrics: ResourceMetrics, done: (result: ExportResult) => void): void {
+    for (const libraryMetrics of metrics.scopeMetrics) {
+      for (const metric of libraryMetrics.metrics) {
+        console.dir(metric.descriptor);
+        console.dir(DataPointType[metric.dataPointType]);
+        for (const dataPoint of metric.dataPoints) {
+          console.dir(dataPoint);
```

Comment:
Do we really want 3 separate console lines? How does the collector log metrics? Maybe we can emulate the format?

Reply:
Not sure but I have changed it into a one liner. I assume this is something we could improve later on?

Reply:
Maybe. Depends if we consider changing the output of the exporter to be a breaking change. Someone could for example use this exporter to get traces out of a lambda function using cloudwatch

Reply:
Hmm,  okay, I will see if I can find an example of collector which outputs metrics console. I don't fully follow the Go code for collector to extract what the format is. Looks like it might be OTLP metrics serialised to JSON